java/ql/test/ext/TopJdkApis/TopJdkApis.qll

```diff
@@ -0,0 +1,103 @@
+      "java.lang.IllegalArgumentException#IllegalArgumentException(String)",
+      "java.lang.Class#getName()", "java.lang.Enum#Enum(String,int)",
+      "java.io.PrintWriter#write(String)", "java.util.Entry#getValue()", "java.util.Entry#getKey()",
+      "java.util.Iterator#next()", "java.lang.Object#hashCode()",
+      "java.util.Optional#orElse(Object)", "java.lang.StringBuffer#append(String)",
+      "java.util.Collections#singletonList(Object)", "java.lang.Iterable#forEach(Consumer)",
+      "java.util.Optional#of(Object)", "java.lang.String#contains(CharSequence)",
+      "java.util.ArrayList#add(Object)", "java.util.Optional#ofNullable(Object)",
+      "java.util.Collections#emptyList()", "java.math.BigDecimal#BigDecimal(String)",
+      "java.lang.System#currentTimeMillis()", "java.lang.Object#equals(Object)",
+      "java.util.Map#containsKey(Object)", "java.util.Optional#isPresent()",
+      "java.lang.String#trim()", "java.util.List#addAll(Collection)",
+      "java.util.Set#contains(Object)", "java.util.Optional#map(Function)",
+      "java.util.Map#entrySet()", "java.util.Optional#empty()",
+      "java.lang.Integer#parseInt(String)", "java.lang.String#startsWith(String)",
+      "java.lang.IllegalStateException#IllegalStateException(String)",
+      "java.lang.Enum#equals(Object)", "java.util.Iterator#hasNext()",
+      "java.util.List#contains(Object)", "java.lang.String#substring(int,int)",
+      "java.util.List#of(Object)", "java.util.Objects#hash(Object[])",
+      "java.lang.RuntimeException#RuntimeException(String)", "java.lang.String#isEmpty()",
+      "java.lang.String#replace(CharSequence,CharSequence)", "java.util.Set#size()",
+      "java.io.File#File(String)", "java.lang.StringBuilder#append(Object)",
+      "java.lang.String#split(String)", "java.util.Map#values()", "java.util.UUID#randomUUID()",
+      "java.util.ArrayList#ArrayList(Collection)", "java.util.Map#keySet()",
+      "java.sql.ResultSet#getString(String)", "java.lang.String#hashCode()",
+      "java.lang.Throwable#Throwable(Throwable)", "java.util.HashMap#get(Object)",
+      "java.lang.Class#getSimpleName()", "java.util.Set#isEmpty()", "java.util.Map#size()",
+      "java.lang.String#substring(int)", "java.util.Map#remove(Object)",
+      "java.lang.Throwable#printStackTrace()", "java.util.stream.Stream#findFirst()",
+      "java.util.Optional#ifPresent(Consumer)", "java.lang.String#valueOf(Object)",
+      "java.lang.String#toLowerCase()", "java.util.UUID#toString()",
+      "java.lang.StringBuilder#append(int)", "java.util.Objects#requireNonNull(Object,String)",
+      "java.nio.file.Path#resolve(String)", "java.lang.Enum#toString()",
+      "java.lang.RuntimeException#RuntimeException(Throwable)", "java.util.Collection#size()",
+      "java.lang.String#charAt(int)", "java.util.stream.Stream#forEach(Consumer)",
+      "java.util.Map#isEmpty()", "java.lang.String#valueOf(int)"
+    ]
+}
+
+predicate hasCallable(string apiName) {
+  exists(Callable c |
```

Comment:
If you make this a parameter of the predicate, then you can reduce some code duplication, as the body of this predicate is repeated 3 times (here, in the definition of TopJdkApi, and in the test).

Reply:
I don't think I'm fully understanding the code structure you have in mind for this. 🤔 
I've added a `string getApiName(Callable api)` predicate in https://github.com/github/codeql/pull/11572/commits/c0628035fadd40774786deb28ee22bc574db4cb4 for now to somewhat reduce the duplication. Let me know if that structure is alright, otherwise I'll think about this some more to see if I can come up with what you are proposing.